services/smtp-bridge-emulator/apihandlers/send-email.go

```diff
@@ -0,0 +1,188 @@
+	Subject         string                          `json:"subject"`
+	Content         string                          `json:"content"`
+	HighPrio        bool                            `json:"highPrio"`
+	HeaderOverrides *messagingTypes.HeaderOverrides `json:"headerOverrides"`
+}
+
+func (h *HttpEndpoints) AddRoutes(rg *gin.RouterGroup) {
+	auth := rg.Group("/")
+
+	auth.POST("/send-email",
+		mw.HasValidAPIKey(h.apiKeys),
+		h.sendEmail)
+}
+
+// creates a directory for recipient to store emails (.eml) and handles errors.
+func createRecipientsDirectory(emailsDir string, recipient string) error {
+
+	recipientDIR := filepath.Join(emailsDir, recipient)
+	if err := os.MkdirAll(recipientDIR, os.ModePerm); err != nil {
+		slog.Error("Error creating directory: "+recipientDIR, slog.String("error", err.Error()))
+		return err
+	}
+	return nil
+}
+
+// saves an email to a .eml file inside recipients directory
+func saveEmailAsEML(email SendEmailReq, emailsDir string) error {
+	for _, recipient := range email.To {
+
+		if err := createRecipientsDirectory(emailsDir, recipient); err != nil {
+			return err
+		}
+		// Generate a unique EML file path
+		emlFilePath, err := getUniqueEMLFilePath(emailsDir, recipient)
+		if err != nil {
+			slog.Error("Error generating EML file path", slog.String("FilePath", emlFilePath), slog.String("error", err.Error()))
+			return err
+		}
+
+		// Create .eml file
+		file, err := os.Create(emlFilePath)
+		if err != nil {
+			slog.Error("Failed to create EML file ", slog.String("path", emlFilePath), slog.String("error", err.Error()))
+			return err
+		}
+		defer file.Close()
+
+		// compose email headers and write to file
+		headers := composeHeaders(email)
+		if err := writeHeadersToFile(file, headers); err != nil {
+			slog.Error("Failed to write headers to file", slog.String("FilePath", emlFilePath), slog.String("error", err.Error()))
+			return err
+		}
+
+		// Write content (body) to file
+		_, err = file.WriteString(email.Content)
+		if err != nil {
+			slog.Error("Failed to write email content", slog.String("error", err.Error()))
+			return err
+		}
+	}
+
+	slog.Info("Email has been saved as EML file successfully")
+	return nil
+}
+
+// constructs the email headers as a map from SendEmailReq
+func composeHeaders(email SendEmailReq) map[string]string {
+	headers := map[string]string{
+		"To":      strings.Join(email.To, ", "),
+		"Subject": email.Subject,
+		"Date":    time.Now().Format(time.RFC1123Z),
+	}
+
+	// Priority header
+	if email.HighPrio {
+		headers["Priority"] = "High"
+	}
+
+	// Optional header overrides
+	if email.HeaderOverrides != nil {
+		if email.HeaderOverrides.From != "" {
+			headers["From"] = email.HeaderOverrides.From
+		}
+		if email.HeaderOverrides.Sender != "" {
+			headers["Sender"] = email.HeaderOverrides.Sender
+		}
+		if !email.HeaderOverrides.NoReplyTo && len(email.HeaderOverrides.ReplyTo) > 0 {
+			headers["Reply-To"] = strings.Join(email.HeaderOverrides.ReplyTo, ", ")
+		}
+	}
+
```

Comment:
I think we need the content type in the header as well, so readers interpret it as html correctly:
```
Content-Type: text/html; charset="UTF-8"
Content-Transfer-Encoding: quoted-printable. (not sure about this one, was in the example I've found)
```

Reply:
Content-Transfer-Encoding seems to be needed to encode smileys or non-ASCII alphabets in the email.